test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,118 @@
-      pod.metadata.ownerReferences && pod.metadata.ownerReferences.first.kind == "ReplicaSet"
+  def test_full_basic_set_deploy_succeeds
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+  end
+
+  def test_partial_deploy_followed_by_full_deploy
+    deploy_fixture_set("basic", ["configmap-data", "redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+    basic.assert_configmap_data_up
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+
+    deploy_fixture_set("basic")
+    basic.assert_all_up
+  end
+
+  def test_pruning
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+
+    deploy_fixture_set("basic", ["redis"])
+    basic.assert_all_redis_resources_up
+    basic.refute_configmap_data_exists
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+  end
+
+  def test_pvcs_are_not_pruned
+    deploy_fixture_set("basic", ["redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+
+    deploy_fixture_set("basic", ["configmap-data"])
+    basic.assert_configmap_data_up
+    basic.refute_redis_resources_exist(expect_pvc: true)
+  end
+
+  def test_success_with_unrecognized_resource_type
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    secret = {
+      "apiVersion" => "v1",
+      "kind" => "Secret",
+      "metadata" => { "name" => "test" },
+      "data" => { "foo" => "YmFy" }
+    }
+    fixture_set["secret"] = { "Secret" => secret }
+    deploy_loaded_fixture_set(fixture_set)
+
+    live_secret = kubeclient.get_secret("test", @namespace)
+    assert_equal({ foo: "YmFy" }, live_secret["data"].to_h)
+  end
+
+  def test_invalid_yaml_fails_fast
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_fixture_set("invalid", ["yaml-error"])
     end
+    assert_match /Template \S+ cannot be parsed/, error.to_s
+  end
+
+  def test_invalid_k8s_spec_that_is_valid_yaml_fails_fast
+    fixture_set = load_fixture_data("basic", ["configmap-data"])
+    configmap = fixture_set["configmap-data"]["ConfigMap"].first
+    configmap["metadata"]["myKey"] = "uhOh"
+
+    error = assert_raises(KubernetesDeploy::FatalDeploymentError) do
+      deploy_loaded_fixture_set(fixture_set)
+    end
+    assert_match /Dry run failed for template configmap-data/, error.to_s
+
+    @logger_stream.rewind
+    assert_match /error validating data\: found invalid field myKey for v1.ObjectMeta/, @logger_stream.read
+  end
+
+  def test_bad_container_image_on_run_once_halts_and_fails_deploy
+    fixture_set = load_fixture_data("basic")
+    pod = fixture_set["unmanaged-pod"]["Pod"].first
+    pod["spec"]["activeDeadlineSeconds"] = 3
+    pod["spec"]["containers"].first["image"] = "hello-world:elephants"
```

Comment:
Do you mean as someone reading the tests (in which case I should change it to `thisImageIsBad`) or are you worried the elephants tag could actually exist in the future (in which case I should do something like append a random string).